Shape_detection/include/CGAL/Shape_detection/Region_growing/Region_growing_on_point_set/Least_squares_circle_fit_region.h

```diff
@@ -0,0 +1,369 @@
+
+#include <vector>
+
+#include <CGAL/assertions.h>
+#include <CGAL/number_utils.h>
+#include <CGAL/Cartesian_converter.h>
+#include <CGAL/Eigen_diagonalize_traits.h>
+
+#include <CGAL/Shape_detection/Region_growing/internal/utils.h>
+
+namespace CGAL {
+namespace Shape_detection {
+namespace Point_set {
+
+/*!
+  \ingroup PkgShapeDetectionRGOnPoints
+
+  \brief Region type based on the quality of the least squares circle
+  fit applied to 2D points.
+
+  This class fits a circle to chunks of points in a 2D point set and
+  controls the quality of this fit.  If all quality conditions are
+  satisfied, the chunk is accepted as a valid region, otherwise
+  rejected.
+
+  \tparam GeomTraits
+  must be a model of `Kernel`.
+
+  \tparam InputRange
+  must be a model of `ConstRange` whose iterator type is `RandomAccessIterator`.
+
+  \tparam PointMap
+  must be an `LvaluePropertyMap` whose key type is the value type of the input
+  range and value type is `Kernel::Point_3`.
+
+  \tparam NormalMap
+  must be an `LvaluePropertyMap` whose key type is the value type of the input
+  range and value type is `Kernel::Vector_3`.
+
+  \cgalModels `RegionType`
+*/
+template<typename GeomTraits,
+         typename InputRange,
+         typename PointMap,
+         typename NormalMap>
+class Least_squares_circle_fit_region
+{
+
+public:
+
+  /// \name Types
+  /// @{
+
+  /// \cond SKIP_IN_MANUAL
+  using Traits = GeomTraits;
+  using Input_range = InputRange;
+  using Point_map = PointMap;
+  using Normal_map = NormalMap;
+  /// \endcond
+
+  /// Number type.
+  typedef typename GeomTraits::FT FT;
+
+  /// \cond SKIP_IN_MANUAL
+  using Point_2 = typename Traits::Point_2;
+  using Vector_2 = typename Traits::Vector_2;
+
+  using Squared_distance_2 = typename Traits::Compute_squared_distance_2;
+
+  using Get_sqrt = internal::Get_sqrt<Traits>;
+  using Sqrt = typename Get_sqrt::Sqrt;
+
+private:
+
+  const Input_range& m_input_range;
+
+  const FT m_distance_threshold;
+  const FT m_normal_threshold;
+  const std::size_t m_min_region_size;
+  const FT m_min_radius;
+  const FT m_max_radius;
+
+  const Point_map m_point_map;
+  const Normal_map m_normal_map;
+
+  const Squared_distance_2 m_squared_distance_2;
+  const Sqrt m_sqrt;
+
+  Point_2 m_center;
+  FT m_radius;
+
+public:
+
+  /// \endcond
+
+  /// @}
+
+  /// \name Initialization
+  /// @{
+
+  /*!
+    \brief initializes all internal data structures.
+
+    \param input_range an instance of `InputRange` with 2D points and
+    corresponding 2D normal vectors
+
+    \param distance_threshold the maximum distance from a point to a
+    circle. %Default is 1.
+
+    \param angle_threshold the maximum accepted angle in degrees
+    between the normal of a point and the radius of a circle. %Default
+    is 25 degrees.
+
+    \param min_region_size the minimum number of 2D points a region
+    must have. %Default is 3.
+
+    \param minimum_radius is the radius below which an estimated
```

Comment:
I would remove `is` to be consistent with other parameters like `angle_threshold` e.g. The same for `maximum_radius` and also for other classes with spheres and cylinders.